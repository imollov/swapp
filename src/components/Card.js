import React from 'react'
import {
  Card as CardContainer,
  CardImg,
  CardBody,
  CardTitle,
  CardSubtitle,
  CardText
} from 'reactstrap'
import styles from './Card.module.scss'

const Card = ({ img, title, subtitle, text, className, onClick, children }) => (
  <CardContainer className={`${styles.card} ${className || ''}`}>
    {img && <CardImg src={img} onClick={onClick} />}
    <CardBody>
      {title && <CardTitle onClick={onClick}>{title}</CardTitle>}
      {subtitle && <CardSubtitle>{subtitle}</CardSubtitle>}
      {text && <CardText>{text}</CardText>}
      {children}
    </CardBody>
  </CardContainer>
)

export default Card
