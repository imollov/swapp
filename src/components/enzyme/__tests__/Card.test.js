import React from 'react'
import { shallow } from 'enzyme'

import Card from '../../Card'

describe('Card Component', () => {
  it('smoke test', () => {
    const wrapper = shallow(<Card />)

    expect(wrapper).toExist()
  })

  it('should match snapshot', () => {
    const wrapper = shallow(
      <Card
        className="class"
        title="Title"
        subtitle="Subtitle"
        text="Text"
        img="img.png"
        onClick={() => {}}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
