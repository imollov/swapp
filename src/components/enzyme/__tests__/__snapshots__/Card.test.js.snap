// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should match snapshot 1`] = `
<Card
  className="card class"
  tag="div"
>
  <CardImg
    onClick={[Function]}
    src="img.png"
    tag="img"
  />
  <CardBody
    tag="div"
  >
    <CardTitle
      onClick={[Function]}
      tag="div"
    >
      Title
    </CardTitle>
    <CardSubtitle
      tag="div"
    >
      Subtitle
    </CardSubtitle>
    <CardText
      tag="p"
    >
      Text
    </CardText>
  </CardBody>
</Card>
`;
